src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,179 @@
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = getNode(index);
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = new Node(value);
+        Node<T> current = getNode(index);
+        T oldValue = current.value;
+        if (index == ZERO) {
+            node.next = current.next;
+            current.next.prev = node;
+            head = node;
+            return oldValue;
+        }
+        if (index == size - ONE) {
+            current.prev.next = node;
+            node.prev = current.prev;
+            tail = node;
+            return oldValue;
+        }
+        node.next = current.next;
+        node.prev = current.prev;
+        current.prev.next = node;
+        current.next.prev = node;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> current = getNode(index);
+        if (index == ZERO && size == ONE) {
+            head = null;
+            tail = null;
+            size--;
+            return current.value;
+        }
+        if (index == ZERO && size != ZERO) {
+            head = current.next;
+            current.next.prev = null;
+            size--;
+            return current.value;
+        }
+        if (index == size - ONE) {
+            tail = current.prev;
+            current.prev.next = null;
+            size--;
+            return current.value;
+        }
+        checkIndex(index);
+        current.prev.next = current.next;
+        current.next.prev = current.prev;
+        size--;
+        return current.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        int index = ZERO;
+        do {
+            if ((current.value == object)
+                    || current.value != null && current.value.equals(object)) {
+                remove(index);
+                return true;
+            }
+            index++;
+            current = current.next;
+
+        } while (current.next != null);
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
     }
+
+    private Node<T> getNode(int index) {
+        checkIndex(index);
+        Node<T> current = head;
+        int i = index;
+        while (current.next != null && i != ZERO) {
+            current = current.next;
+            i--;
+        }
+        return current;
+    }
+
+    private void checkIndex(int i) {
+        if (i >= size || i < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    private static class Node<T> {
+        private Node next;
+        private Node prev;
+        private T value;
+
+        public Node(T value) {
```

Comment:
I guess if you use the constructor  with all fields  it will simplify  your solution